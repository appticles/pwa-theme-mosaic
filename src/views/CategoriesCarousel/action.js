import config from 'react-global-configuration';

export const REQUEST_CATEGORIES = 'REQUEST_CATEGORIES';
export const RECEIVE_CATEGORIES = 'RECEIVE_CATEGORIES';

export const requestCategories = () => ({
  type: REQUEST_CATEGORIES,
});

export const receiveCategories = categories => ({
  type: RECEIVE_CATEGORIES,
  categories,
});

export const fetchCategories = (params = {}) => (dispatch) => {
  dispatch(requestCategories());

  let url;
  if (params && params.post) {
    url = config.get('export').categories + '?post=' + String(params.post);
  } else {
    url =
      config.get('export').categories +
      '?' +
      Object.keys(params)
        .map(k => k + '=' + encodeURIComponent(params[k]))
        .join('&');
  }
  return fetch(url)
    .then(response => response.json())
    .then(json => dispatch(receiveCategories(json)))
    .catch(() => {
      dispatch(receiveCategories([]));
    });
};
