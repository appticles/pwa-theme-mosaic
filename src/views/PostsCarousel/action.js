import config from 'react-global-configuration';

export const REQUEST_POSTS = 'REQUEST_POSTS';
export const RECEIVE_POSTS = 'RECEIVE_POSTS';

export const requestPosts = () => ({
  type: REQUEST_POSTS,
});

export const receivePosts = posts => ({
  type: RECEIVE_POSTS,
  posts,
});

export const fetchPosts = (params = {}) => (dispatch) => {
  dispatch(requestPosts());
  config.set(window.__INITIAL_CONFIG__, { freeze: false });

  let url;
  if (params && params.id) {
    url = config.get('export').posts + '/' + String(params.id);
  } else {
    url =
      config.get('export').posts +
      '?' +
      Object.keys(params)
        .map(k => k + '=' + encodeURIComponent(params[k]))
        .join('&');
  }
  return fetch(url)
    .then(response => response.json())
    .then(json => dispatch(receivePosts(json)))
    .catch(() => {
      dispatch(receivePosts([]));
    });
};
