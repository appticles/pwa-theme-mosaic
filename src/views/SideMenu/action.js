import config from 'react-global-configuration';

export const REQUEST_PAGES = 'REQUEST_PAGES';
export const RECEIVE_PAGES = 'RECEIVE_PAGES';

export const requestPages = () => ({
  type: REQUEST_PAGES,
});

export const receivePages = pages => ({
  type: RECEIVE_PAGES,
  pages,
});

export const fetchPages = (params = {}) => (dispatch) => {
  dispatch(requestPages());

  let url;
  if (params && params.id) {
    url = config.get('export').pages + '/' + String(params.id);
  } else {
    url =
      config.get('export').pages +
      '?' +
      Object.keys(params)
        .map(k => k + '=' + encodeURIComponent(params[k]))
        .join('&');
  }
  return fetch(url)
    .then(response => response.json())
    .then(json => dispatch(receivePages(json)))
    .catch(() => {
      dispatch(receivePages([]));
    });
};
