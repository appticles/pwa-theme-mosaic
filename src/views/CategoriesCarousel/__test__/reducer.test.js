import categories, { INITIAL_STATE, getCategories, getCategoriesFetching } from '../reducer';
import { requestCategories, receiveCategories } from '../action';

describe('Categories reducer', () => {
  it('returns the same state on an unhandled action', () => {
    expect(categories(INITIAL_STATE, { type: '_NULL' })).toMatchSnapshot();
  });
  it('handles REQUEST_CATEGORIES action', () => {
    const categoriesList = {
      isFetching: 0,
      items: [
        {
          id: 1,
          name: 'Nice one',
        },
        {
          id: 2,
          name: 'Nicer one',
        },
      ],
    };
    expect(categories(categoriesList, requestCategories())).toMatchSnapshot();
  });
  it('handles RECEIVE_CATEGORIES action', () => {
    const categoriesState = {
      isFetching: 1,
      items: [
        {
          id: 1,
          name: 'Nice one',
        },
        {
          id: 2,
          name: 'Nicer one',
        },
      ],
    };
    const categoriesList = [
      {
        id: 3,
        name: 'Nicest one',
      },
    ];
    expect(categories(categoriesState, receiveCategories(categoriesList))).toMatchSnapshot();
  });

  // reducer functions
  it('getCategoriesFetching returns state.isFetching', () => {
    const state = {
      isFetching: 4,
    };
    expect(getCategoriesFetching(state)).toBe(4);
  });
  it('getCategories returns state.items', () => {
    const state = {
      items: [
        {
          name: 'Nice one',
        },
      ],
    };
    expect(getCategories(state)).toBe(state.items);
  });
});
