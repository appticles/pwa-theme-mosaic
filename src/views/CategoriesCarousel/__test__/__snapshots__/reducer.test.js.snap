// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories reducer handles RECEIVE_CATEGORIES action and adds new categories to the state 1`] = `
Object {
  "isFetching": 0,
  "items": Array [
    Object {
      "id": 1,
      "name": "Nice one",
    },
    Object {
      "id": 2,
      "name": "Nicer one",
    },
    Object {
      "id": 3,
      "name": "Nicest one",
    },
  ],
  "loadMore": true,
}
`;

exports[`Categories reducer handles RECEIVE_CATEGORIES action and adds new category to the state 1`] = `
Object {
  "isFetching": 0,
  "items": Array [
    Object {
      "id": 1,
      "name": "Nice one",
    },
    Object {
      "id": 2,
      "name": "Nicer one",
    },
    Object {
      "id": 3,
      "name": "Nicest one",
    },
  ],
  "loadMore": true,
}
`;

exports[`Categories reducer handles RECEIVE_CATEGORIES action and continues load more 1`] = `
Object {
  "isFetching": 0,
  "items": Array [
    Object {
      "id": 1,
      "name": "Nice one",
    },
    Object {
      "id": 2,
      "name": "Nicer one",
    },
    Object {
      "id": 3,
      "name": "Nicest one",
    },
    Object {
      "id": 4,
      "name": "Another category",
    },
  ],
  "loadMore": true,
}
`;

exports[`Categories reducer handles RECEIVE_CATEGORIES action and disables load more 1`] = `
Object {
  "isFetching": 0,
  "items": Array [
    Object {
      "id": 1,
      "name": "Nice one",
    },
    Object {
      "id": 2,
      "name": "Nicer one",
    },
    Object {
      "id": 3,
      "name": "Nicest one",
    },
  ],
  "loadMore": false,
}
`;

exports[`Categories reducer handles REQUEST_CATEGORIES action 1`] = `
Object {
  "isFetching": 1,
  "items": Array [
    Object {
      "id": 1,
      "name": "Nice one",
    },
    Object {
      "id": 2,
      "name": "Nicer one",
    },
  ],
  "loadMore": true,
}
`;

exports[`Categories reducer returns the same state on an unhandled action 1`] = `
Object {
  "isFetching": 0,
  "items": Array [],
  "loadMore": true,
}
`;
